lib/modules/manager/scalafmt/extract.ts

```diff
@@ -0,0 +1,64 @@
+import { logger } from '../../../logger';
+import { readLocalFile } from '../../../util/fs';
+import { regEx } from '../../../util/regex';
+import { GithubReleasesDatasource } from '../../datasource/github-releases';
+import * as semverVersioning from '../../versioning/semver';
+import type {
+  ExtractConfig,
+  PackageDependency,
+  PackageFile,
+  PackageFileContent,
+} from '../types';
+
+const scalafmtVersionRegex = regEx(
+  'version *= *(?<version>\\d+\\.\\d+\\.\\d+)',
+);
+
+export function extractPackageFile(
+  content: string,
+  packageFile: string,
+): PackageFileContent | null {
+  const regexResult = scalafmtVersionRegex.exec(content);
+  const scalafmtVersion = regexResult?.groups?.version;
+  const matchString = regexResult?.[0];
+  if (scalafmtVersion) {
+    const scalafmtDependency: PackageDependency = {
+      datasource: GithubReleasesDatasource.id,
+      depName: 'scalameta/scalafmt',
+      packageName: 'scalameta/scalafmt',
```

Comment:
```suggestion
```
Only one of `depName` or `packageName` are needed if they contain the same value. 
The first should be human-readable and the later is the name which is actually looked up by the datasource.

Reply:
I kept "scalafmt" as human-readable name then, that's how most people know the tool.